src/core/manager.c

```diff
@@ -3740,8 +3742,45 @@ static int build_generator_environment(Manager *m, char ***ret) {
         return 0;
 }
 
+static int manager_execute_generators(Manager *m, char **paths) {
+        _cleanup_strv_free_ char **ge = NULL;
+        const char *argv[] = {
+                NULL, /* Leave this empty, execute_directory() will fill something in */
+                m->lookup_paths.generator,
+                m->lookup_paths.generator_early,
+                m->lookup_paths.generator_late,
+                NULL,
+        };
+        int r;
+
+        r = build_generator_environment(m, &ge);
+        if (r < 0)
+                return log_error_errno(r, "Failed to build generator environment: %m");
+
+        /* We leave /run read-write because that's where the output goes to,
```

Comment:
why the whole of /run/? wouldn't /run/systemd/ suffice?

Reply:
I don't think we should cut of /run, for the same reason that we leave /dev/shm: /run is expected to be writeable. Right now I can just leave /run by not remounting it, and to remount it ro and then punch holes is a lot more code.